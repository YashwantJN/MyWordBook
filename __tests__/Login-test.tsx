/**
 * @format
 */

import 'react-native';
import React from 'react';
import renderer, {act} from 'react-test-renderer';
import LoginScreen from '../src/modules/auth/LogIn';
// import {fireEvent, render} from '@testing-library/react-native';

jest.mock('@react-navigation/native', () => {
  return {
    ...jest.requireActual('@react-navigation/native'),
    useRoute: jest.fn(() => ({
      params: {
        user: {email: ''},
      },
    })),
  };
});

describe('LoginScreen', () => {
  test('login renders correctly', async () => {
    const tree = renderer.create(<LoginScreen />).toJSON();
    await act(async () => {
      expect(tree).toMatchSnapshot();
    });
  });
});
