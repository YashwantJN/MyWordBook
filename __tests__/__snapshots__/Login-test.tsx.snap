// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginScreen login renders correctly 1`] = `
<RCTScrollView
  automaticallyAdjustKeyboardInsets={true}
  indicatorStyle="black"
  keyboardShouldPersistTaps="handled"
  showsVerticalScrollIndicator={true}
>
  <View>
    <View
      onLayout={[Function]}
      style={
        Object {
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <View
        style={
          Object {
            "height": 1334,
            "justifyContent": "center",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#183d3d",
              "fontSize": 24,
              "fontWeight": "700",
              "letterSpacing": 1,
              "marginBottom": 20,
              "paddingBottom": 10,
              "textAlign": "center",
            }
          }
        >
          Authenticate Here!
        </Text>
        <View
          style={
            Object {
              "height": 400,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "white",
                  "borderRadius": 30,
                  "flex": 1,
                  "justifyContent": "space-evenly",
                  "margin": 20,
                },
              ]
            }
          >
            <View>
              <Text
                style={
                  Object {
                    "alignSelf": "center",
                    "color": "#183d3d",
                    "fontSize": 18,
                    "fontWeight": "500",
                    "width": "80%",
                  }
                }
              >
                Email
              </Text>
              <TextInput
                autoCapitalize="none"
                autoComplete="off"
                autoCorrect={false}
                keyboardType="default"
                onChangeText={[Function]}
                placeholder="Enter your email"
                placeholderTextColor="gray"
                secureTextEntry={false}
                style={
                  Object {
                    "alignSelf": "center",
                    "backgroundColor": "white",
                    "borderBottomWidth": 0.5,
                    "borderRadius": 10,
                    "color": "black",
                    "fontSize": 18,
                    "height": 44,
                    "width": "80%",
                  }
                }
                testID="Email"
                value=""
              />
              <View
                style={
                  Object {
                    "height": 25,
                    "justifyContent": "center",
                  }
                }
              />
            </View>
            <View>
              <Text
                style={
                  Object {
                    "alignSelf": "center",
                    "color": "#183d3d",
                    "fontSize": 18,
                    "fontWeight": "500",
                    "width": "80%",
                  }
                }
              >
                Password
              </Text>
              <TextInput
                autoCapitalize="none"
                autoComplete="off"
                autoCorrect={false}
                keyboardType="default"
                onChangeText={[Function]}
                placeholder="Enter your password"
                placeholderTextColor="gray"
                secureTextEntry={true}
                style={
                  Object {
                    "alignSelf": "center",
                    "backgroundColor": "white",
                    "borderBottomWidth": 0.5,
                    "borderRadius": 10,
                    "color": "black",
                    "fontSize": 18,
                    "height": 44,
                    "width": "80%",
                  }
                }
                testID="Password"
                value=""
              />
              <View
                style={
                  Object {
                    "height": 25,
                    "justifyContent": "center",
                  }
                }
              />
            </View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "height": 54,
                  "justifyContent": "center",
                }
              }
            >
              <View
                accessible={true}
                collapsable={false}
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#183d3d",
                    "borderRadius": 12,
                    "elevation": 3,
                    "flex": 1,
                    "justifyContent": "center",
                    "width": "80%",
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "white",
                      "fontSize": 18,
                      "fontWeight": "700",
                      "textAlign": "center",
                    }
                  }
                >
                  Submit
                </Text>
              </View>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "height": 40,
            }
          }
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "flex-end",
                "borderRadius": 12,
                "justifyContent": "flex-start",
                "right": 34,
                "textAlign": "right",
                "top": 0,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#035efc",
                  "fontSize": 16,
                  "fontWeight": "500",
                }
              }
            >
              Forgot Password?
            </Text>
          </View>
        </View>
        <Modal
          animationType="none"
          hardwareAccelerated={false}
          onRequestClose={[Function]}
          statusBarTranslucent={true}
          style={
            Object {
              "zIndex": 1100,
            }
          }
          transparent={true}
          visible={false}
        />
      </View>
    </View>
  </View>
</RCTScrollView>
`;
